### DISCUSSION TITLE
Set background color for ressource in resourceTimeGridWeek

### DISCUSSION THREAD

**Discussion by Retrovers** [vkurko/calendar#354] [General]
Hello,
Is it possible to set a background color for each resource like this in resourceTimeGridWeek view:
![Capture d’écran 2024-10-17 à 15 17 35](https://github.com/user-attachments/assets/dccd474c-2c24-45f7-9ce4-b83969917062)

Thank you for your amazing work!

**Comment by vkurko** [comment]
Unfortunately there is no such setting. You can use the [background](https://github.com/vkurko/calendar?tab=readme-ov-file#parsing-event-from-a-plain-object) events for the whole week to change the background color.

**Comment by Retrovers** [comment]
Yes but the "background" option of an event is not used to color the grid of an entire resource?

otherwise I can always try to target the body associated with the resource in JS to put a color there. but this solution is not very clean...

**Comment by JamieCee20** [comment]
Just want to revive this, as I also have a similar instance where I have 'resourceTimelineWeek' view, but certain resources I want to block out a full back via a background colour, which has no event related to that resource but I cant block it out currently without the ability to set a background colour for the resource

**Comment by Uhasith** [comment]
I want to set the resource background color to red to indicate that it is not available. However, I cannot access the current day inside resourceLabelContent or resourceLabelDidMount. Any workaround? @Retrovers @JamieCee20 

I’m looking for an approach like this:

```
const currentDate = info.currentStart;
const dayName = currentDate
    .toLocaleDateString('en-US', { weekday: 'long' })
    .toLowerCase();

resources.value.forEach((resource: any) => {
    if (resource.offDays?.includes(dayName)) {
        // set background color here
    }
});
```
